File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
  def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
    slow = head
    fast = head.next
    while(slow and fast and fast.next and fast.next.next):
      slow = slow.next
      fast = fast.next.next
    if slow.next == None:
      return None
    slow.next = slow.next.next
    return head

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
  ListNode* deleteMiddle(ListNode* head) {
    if(head->next){
      ListNode* fast = head->next;
      ListNode* slow = head;
      while(slow && fast && fast->next && fast->next->next){
        fast = fast->next->next;
        slow = slow->next;
      }
      slow->next = slow->next->next;
      return head;     
    }
    return nullptr;
  }
};